### DISCUSSION TITLE
How to use edata.visits in IMPRESSION events

### DISCUSSION THREAD

**Discussion by coolbung** [Sunbird-Obsrv/Community#34] [Documentation]
We're using IMPRESSION events to track page view activity. What are the use cases intended for using the `edata.visits` array ? How is this different from the `edata.uri` attribute ?

Below is the edata for IMPRESSION
```json
{
  "edata": {
    "type": "", // Required. Impression type (list, detail, view, edit, workflow, search)
    "subtype": "", // Optional. Additional subtype. "Paginate", "Scroll"
    "pageid": "", // Required. Unique page id
    "uri": "", // Required. Relative URL of the content
    "visits": [VISIT] // Optional. Capture the object visits
  }
}
```

Is generating a single IMPRESSION event for each visit not the recommended approach ?

@anandp504

**Comment by coolbung** [comment]
Can you share inputs @anandp504 @santhoshvasabhaktula ?

**Comment by SanthoshVasabhaktula** [comment]
@coolbung - visits are used to track the "objects" viewed in the page. 

For ex: A search query would have resulted in showing "N" content objects. The "visits" can capture the content objects that are visible to the user in the given resolution. A user would have selected a content or scrolled to the next page. All the events from search -> impression -> interact would have the same correlation id.

This information would have been then used to:
1. Optimize the search engine to provide better results.
2. Optimize the UX for better engagement by the users.
